[fix #7465] Add whatsnew page for Firefox 69

craigcook opened the pull request:
## Description
Adds a whatsnew page for Firefox 69. This is the first step toward establishing the modular framework for this page going forward, though it was being blocked by landing some updates in Protocol first and we couldn't wait for that, so it's initially yet another snowflake page. But Protocol components are in progress and I'll refactor this page when they're ready.

There will also be impending cleanup of the outdated templates and replacing the legacy whatsnew template with a new base. So there's more to be done in the near future, the priority here is getting the needed page in production for Fx69.

## Issue / Bugzilla link
#7465 

## Testing
https://www-demo1.allizom.org/firefox/69.0/whatsnew/

Strings were sent out a few weeks ago so be sure I haven't broken any in the interim. 
https://github.com/mozilla-l10n/www.mozilla.org/blob/master/en-US/firefox/whatsnew_69.lang

There were recent changes to the Monitor button, and this page has two buttons so I've added IDs and should be initializing only the visible button. Did I do that right?

There have also been changes to standardize our analytics params. Did I do that right?

- [ ] New page displays for version 69.*.
- [x] No regressions for previous versions.
- [ ] Strings match what was previously exposed for l10n, with no new strings.
- [x] Monitor button doesn't double-load flow ID.
- [x] Analytics UTM params are correct.

alexgibson commented on bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx69.html:
```diff
@@ -0,0 +1,104 @@
+
+  {{ monitor_button(
+    button_id='monitor-button-signed-in',
+    entrypoint='mozilla.org-firefox-whatsnew69',
```
Rather than repeating values for things like `entrypoint`, `utm_campaign` etc, suggest to reference them using variables for reuse in the view? (avoids things like typos etc).

alexgibson commented on media/static-bundles.json:
```diff
@@ -1065,6 +1065,17 @@
     },
+    {
+      "files": [
+        "js/base/uitour-lib.js",
```
No need to reinclude `uitour-libs.js` or `up-to-date.js` since they are already included in the `firefox_whatsnew_2019` bundle.

craigcook commented on bedrock/firefox/templates/firefox/whatsnew/whatsnew-2019.html:
```diff
@@ -0,0 +1,79 @@
+  # License, v. 2.0. If a copy of the MPL was not distributed with this
+  # file, You can obtain one at http://mozilla.org/MPL/2.0/. -#}
+
+{% add_lang_files "firefox/whatsnew_2019" %}
```
You're right, the evergreen hasn't been exposed yet. I need to do that...

alexgibson replied:
Thinking about it, I'm also not quite clear how this inperplays with templates that extend it (such as `whatsnew-69.html`). If they both use different lang files, does one override the other? Or does activation status kick in from both?

craigcook replied:
Hmm... good question. I think we may need a base template with no content, and the generic/evergreen/legacy page would extend that base.

craigcook replied:
Removed this template, but we'll need to sort this out with #7370

craigcook commented on bedrock/firefox/templates/firefox/whatsnew/whatsnew-2019.html:
```diff
@@ -0,0 +1,79 @@
+      <div class="c-emphasis-box show-fxa-default">
+        <h2 class="c-emphasis-box-title">{{ _('Get the respect you deserve') }}</h2>
+        <p>
+        {% trans promise='#' %}
```
Oh right, this proved to be a little tricky since there are three URLs for three blog posts. For the accounts page we did it in the view and we should do the same here, but this view is much more complicated and I had trouble getting the context right so I punted and forgot to go back to it. I'll work on this.

craigcook replied:
Removed this template

craigcook commented:
> I'm a little confused as to the longer term L10n plan for this new modular page. None of the evergreen content seems to have been exposed for translation yet, which means everything that's translated currently is in `whatsnew_69.lang` (which suggests all that content is only living for one release cycle?). What happens if/when they want to reuse one of the existing components, or do they expect to rewrite copy & content for each release?
> 
> My impression from past discussions was that we would be trying to avoid doing the whole L10n dance every release, and would be instead moving to a more evergreen page that wasn't tied to a specific Firefox release?

As I understand it, content will be updated regularly, for almost every release, as it always has been, we're just settling on one layout and design so it won't be a new from-scratch page every time. Content will still change regularly, but not the design.

The intent behind the "evergreen" content is to have a version-agnostic fallback for old Firefoxes and/or locales that don't get the updated version-specific content. We'll still be doing the l10n dance every release, but probably for a smaller subset of target locales. So it'll be `if_lang_file_active` to get the new page, else it's fallback.

But that evergreen content hasn't been exposed for l10n yet so it won't be available in this release cycle anyway, all of that will be part of the next step to clean up all the old templates and establish the new base template. That's a level of complexity that I didn't want to pile into one PR just to get this page out for 69, so I'm going to do all that debt paydown separately, probably next week (#7370 )

So... I think I'll rework this a bit so it really will be just adding the new page for 69 without trying to extend a new base template or add any evergreen strings without an evergreen page to fall back to.

craigcook commented on bedrock/firefox/views.py:
```diff
@@ -614,6 +614,8 @@ def get_template_names(self):
                 template = 'firefox/whatsnew/index.html'
         elif locale == 'id':
             template = 'firefox/whatsnew/index-lite.id.html'
+        elif version.startswith('69.'):
```
Sorry, I meant `69.*`, so 69.0.1 etc. Fx 70 or greater would just get the legacy page (for now, at least until we add a new rule for 70.*).

craigcook commented on media/css/firefox/whatsnew/whatsnew-2019.scss:
```diff
@@ -75,71 +75,76 @@ $image-path: '/media/protocol/img';
+    box-shadow: 0 4px 16px transparentize($color-off-black, .76), 0 0 2px transparentize($color-off-black, .9);
     margin: 0 auto $layout-sm;
     max-width: $content-md - ($layout-md * 2);
     padding: $spacing-xl $layout-md;
```
The left-aligned button text is coming from Protocol, which doesn't declare an alignment for `.mzp-c-button` so it's just getting the standard text alignment. The global form styles in Protocol _do_ declare `text-align: center` for `button` elements but this is a link so it's not getting that centering.

We should fix this upstream and center-align all button atoms.

craigcook commented on bedrock/firefox/templates/firefox/whatsnew/whatsnew-2019.html:
```diff
@@ -0,0 +1,79 @@
+    <div class="mzp-l-content">
+    {% block content_main %}
+      <div class="c-emphasis-box show-fxa-default">
+        <h2 class="c-emphasis-box-title">{{ _('Get the respect you deserve') }}</h2>
```
Removed template

craigcook commented:
Updated and simplified a bit. I haven't put the analytics values into variables, though I do think that's a good idea, just not sure how best to go about it.

Also apparently [the dark theme performed better](https://github.com/mozilla/bedrock/issues/7465#issuecomment-523209957), so this page is dark now.

ejregithub commented:
@craigcook - please apply your Estimate here.
